jupyter_client/client.py

```diff
@@ -337,11 +337,11 @@ def stop_channels(self) -> None:
     def channels_running(self) -> bool:
         """Are any of the channels created and running?"""
         return (
-            self.shell_channel.is_alive()
-            or self.iopub_channel.is_alive()
-            or self.stdin_channel.is_alive()
-            or self.hb_channel.is_alive()
-            or self.control_channel.is_alive()
+            (self._shell_channel and self.shell_channel.is_alive())
+            or (self._iopub_channel and self.iopub_channel.is_alive())
+            or (self._stdin_channel and self.stdin_channel.is_alive())
+            or (self._hb_channel and self.hb_channel.is_alive())
+            or (self._control_channel and self.control_channel.is_alive())
```

Comment:
These changes seem more in line with the docstring `"Are any of the channels created and running?"`, e.g. with only `self.shell_channel.is_alive()` we are creating a shell channel if it doesn't already exists, so we are only checking if it is running.
Your changes ensure that a channel already exists before checking if it is running, which seems like the right thing to do IMO.

Reply:
Thanks @davidbrochart!